db/database.go

```diff
@@ -492,6 +520,9 @@ func (context *DatabaseContext) Close() {
 	context.mutationListener.Stop()
 	context.changeCache.Stop()
 	context.ImportListener.Stop()
+	if context.Heartbeater != nil {
+		context.Heartbeater.Stop()
+	}
```

Comment:
Already doing nil checks inside stop, probably safer/easier to use if it's down at that level.

```suggestion
	context.Heartbeater.Stop()
```

Reply:
That was my first thought as well, but because context.Heartbeater is defined as Heartbeater interface, this throws a NPE if it's nil.  I came close to just scrapping the interface altogether (since we only have a single implementation), but left it in as it will be useful to have it abstracted if we ever get ns_server integration.